### PR TITLE
Add IBM VPC Block CSI Driver Operator

### PR DESCRIPTION
**Pull request by dobsonj**
Epic: https://issues.redhat.com/browse/STOR-598
I was able to get the driver and operator deployed with these changes and verify that PVC provisioning works:
```
[jon@dagobah ibm]$ oc get pods -n openshift-cluster-csi-drivers
NAME                                                 READY   STATUS    RESTARTS   AGE
ibm-vpc-block-csi-controller-5548dcc7b7-wc9z4        5/5     Running   0          3m10s
ibm-vpc-block-csi-driver-operator-6cdd5d465f-nsrlg   1/1     Running   0          3m16s
ibm-vpc-block-csi-node-4bp86                         3/3     Running   0          2m42s
ibm-vpc-block-csi-node-bgrjn                         3/3     Running   0          3m9s
ibm-vpc-block-csi-node-fxvvk                         3/3     Running   0          2m27s
ibm-vpc-block-csi-node-g57sz                         3/3     Running   0          2m12s
ibm-vpc-block-csi-node-nrsnb                         3/3     Running   0          119s
ibm-vpc-block-csi-node-zsrhw                         3/3     Running   0          2m57s
[jon@dagobah ibm]$ oc create -f validPVC.yaml
persistentvolumeclaim/csi-block-pvc-good created
[jon@dagobah ibm]$ oc create -f validPOD.yaml
pod/ubuntu created
[jon@dagobah ibm]$ oc get pods
NAME     READY   STATUS    RESTARTS   AGE
ubuntu   1/1     Running   0          47s
[jon@dagobah ibm]$ oc get pvc
NAME                 STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                AGE
csi-block-pvc-good   Bound    pvc-02efd13d-a3f1-4e60-bd79-5625ec3eeed0   10Gi       RWO            ibmc-vpc-block-5iops-tier   67s
[jon@dagobah ibm]$ oc get pv
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                        STORAGECLASS                REASON   AGE
pvc-02efd13d-a3f1-4e60-bd79-5625ec3eeed0   10Gi       RWO            Delete           Bound    default/csi-block-pvc-good   ibmc-vpc-block-5iops-tier            48s
```
/cc @openshift/storage

### CODE REVIEW

File: pkg/operator/csidriveroperator/csioperatorclient/ibm-vpc-block.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	pairs := []string{
+		"${OPERATOR_IMAGE}", os.Getenv(envIBMVPCBlockDriverOperatorImage),
+		"${DRIVER_IMAGE}", os.Getenv(envIBMVPCBlockDriverImage),
+		"${NODE_LABEL_IMAGE}", os.Getenv(envIBMVPCNodeLabelUpdaterImage),
+	}
+
+	return CSIOperatorConfig{
+		CSIDriverName:   IBMVPCBlockCSIDriverName,
+		ConditionPrefix: "IBMVPCBlock",
+		Platform:        configv1.IBMCloudPlatformType,
```
Comment (dobsonj) [inline pkg/operator/csidriveroperator/csioperatorclient/ibm-vpc-block.go:27]: I need to come back to this. The operator is only for VPC clusters, but `IBMCloudPlatformType` doesn't distinguish between VPC and Classic. We'll have to look at the platform status and check for `IBMCloudProviderTypeVPC`:
https://github.com/openshift/api/blob/9b3afd95c93af476bec246b2c60f64235f078198/config/v1/types_infrastructure.go#L174-L187

**Comment by jsafrane** [comment]
lgtm-ish

**Comment by dobsonj** [comment]
The StatusFilter I added wasn't working because `ProviderType` is not actually set:
```
"platform":"IBMCloud","platformStatus":{"ibmcloud":{"cisInstanceCRN":"crn:v1:bluemix:public:internet-svcs:global:a/18240e5ed3f647cb96ca52dc92d9addd:f32087fe-9e0d-44a6-9180-0be8360a8435::","location":"us-south","resourceGroupName":"jdobson-rg01"},"type":"IBMCloud"}}}]
```
I'll remove the StatusFilter code. IPI installs only support VPC, and apparently ROKS does not use CSO. If that ever changes, we can address it in a later PR.

**Comment by dobsonj** [comment]
/retitle Add IBM VPC Block CSI Driver Operator

File: assets/csidriveroperators/ibm-vpc-block/06_clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  - list
+  - watch
+  - update
+  - delete
+- apiGroups:
+  - cloudcredential.openshift.io
+  resources:
+  - credentialsrequests
+  verbs:
+  - '*'
```
Comment (bertinatto) [inline assets/csidriveroperators/ibm-vpc-block/06_clusterrole.yaml:253]: Do you need to create a CredentialsRequest object too?

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Comment by bertinatto** [comment]
retest-required

**Pull request merged**